src/registry_schemas/schemas/share_structure.json

```diff
@@ -90,7 +90,6 @@
       ]
     }
   },
-  "required": ["shareClasses"],
```

Comment:
I did not find a way to remove this only for alteration. This change will impact incorporationApplication, transition and conversion. Is there a way to handle in schema? If not I have to add additional validation in legal-api

Reply:
I was thinking we would update validation in legal api.  but not sure what others think of this approach

Reply:
It could have been done easily if we had the reference to the filing type in the share structure.  My initial thought was to pull it out of the share structure as the resolution dates doesnt seem to be tied to a share structure change always. That will have a bigger impact (UI, API, filer etc). So I am fine with this approach and we can validate in the legal api.

Legal API has a method has_at_least_one_share_class that gets called in case of alteration and correction. I did not check whether it is used in all other filing types you mentioned. It definitely has to be removed from alteration and has to be added for those filing type validations where share classes is mandatory

Reply:
Changed this approach. Now it should not impact other filing schema

Reply:
The reason I dint suggest this is I am not sure whether alteration is the only filing type that will pass resolution dates but no share classes. If thats the case this is fine. We dont want to put this in other filing types as well. Linda should be able to confirm that

Reply:
Reply from Linda: _Right now it is just Alterations. However, we will build a correction filing which will need that functionality. The other filings with share structures are: incorporations/continuations/amalgamations and these are filings where you create share structures, not change them._